WalletWasabi/Transactions/AllTransactionStore.cs

```diff
@@ -0,0 +1,236 @@
+				var confirmedWorkFolder = Path.Combine(WorkFolderPath, "ConfirmedTransactions");
+
+				var initTasks = new[]
+				{
+					MempoolStore.InitializeAsync(mempoolWorkFolder, Network, $"{nameof(MempoolStore)}.{nameof(MempoolStore.InitializeAsync)}"),
+					ConfirmedStore.InitializeAsync(confirmedWorkFolder, Network, $"{nameof(ConfirmedStore)}.{nameof(ConfirmedStore.InitializeAsync)}")
+				};
+
+				await Task.WhenAll(initTasks).ConfigureAwait(false);
+				EnsureConsistency();
+
+				if (ensureBackwardsCompatibility)
+				{
+					TryEnsureBackwardsCompatibility();
+				}
+			}
+		}
+
+		private void EnsureConsistency()
+		{
+			lock (Lock)
+			{
+				var mempoolTransactions = MempoolStore.GetTransactionHashes();
+				foreach (var hash in mempoolTransactions)
+				{
+					// Contains is fast, so do this first.
+					if (ConfirmedStore.Contains(hash)
+						&& MempoolStore.TryRemove(hash, out SmartTransaction uTx)
+						&& ConfirmedStore.TryGetTransaction(hash, out SmartTransaction cTx))
+					{
+						cTx.Update(uTx);
+					}
+				}
+			}
+		}
+
+		private void TryEnsureBackwardsCompatibility()
+		{
+			try
+			{
+				// Before Wasabi 1.1.7
+				var networkIndependentTransactionsFolderPath = Path.Combine(EnvironmentHelpers.GetDataDir(Path.Combine("WalletWasabi", "Client")), "Transactions");
+				if (Directory.Exists(networkIndependentTransactionsFolderPath))
+				{
+					var oldTransactionsFolderPath = Path.Combine(networkIndependentTransactionsFolderPath, Network.Name);
+					if (Directory.Exists(oldTransactionsFolderPath))
+					{
+						lock (Lock)
+						{
+							foreach (var filePath in Directory.EnumerateFiles(oldTransactionsFolderPath))
+							{
+								try
+								{
+									string jsonString = File.ReadAllText(filePath, Encoding.UTF8);
+									var allWalletTransactions = JsonConvert.DeserializeObject<IEnumerable<SmartTransaction>>(jsonString)?.OrderByBlockchain() ?? Enumerable.Empty<SmartTransaction>();
+									foreach (var tx in allWalletTransactions)
+									{
+										TryAddNoLock(tx);
+									}
+
+									// ToDo: Uncomment when PR is finished.
+									// File.Delete(filePath);
+								}
+								catch (Exception ex)
+								{
+									Logger.LogTrace(ex);
+								}
+							}
+
+							// ToDo: Uncomment when PR is finished.
+							// Directory.Delete(oldTransactionsFolderPath, recursive: true);
+						}
+					}
+
+					// ToDo: Uncomment when PR is finished.
+					//// If all networks successfully migrated, too, then delete the transactions folder, too.
+					//if (!Directory.EnumerateFileSystemEntries(networkIndependentTransactionsFolderPath).Any())
+					//{
+					//	Directory.Delete(networkIndependentTransactionsFolderPath, recursive: true);
+					//}
+				}
+			}
+			catch (Exception ex)
+			{
+				Logger.LogWarning("Backwards compatibility could not be ensured.");
+				Logger.LogWarning(ex);
+			}
+		}
+
+		private void TryAddNoLock(SmartTransaction tx)
```

Comment:
There are a couple of `TryXxxxx` methods that return `void`. I would expect to get a `bool` because in that way one can take a decision based on that.

Reply:
Because it's often just partial success, but I agree, it's no good like this. I'll remove the tries.